scene/gui/control.cpp

```diff
@@ -703,8 +693,9 @@ void Control::_update_canvas_item_transform() {
 
 Transform2D Control::get_transform() const {
 	ERR_READ_THREAD_GUARD_V(Transform2D());
-	Transform2D xform = _get_internal_transform();
-	xform[2] += get_position();
+	// M = T(position) * T(pivot_offset) * R(rotation) * S(scale) * T(-pivot_offset)
+	Transform2D xform(data.rotation, data.scale, 0.0f, data.pos_cache + data.pivot_offset);
```

Comment:
Using `data.pos_cache` directly instead of `get_position()`, the difference is `get_position()` has additional thread guard. But there's already one in `get_transform`.